File: Java/0003.java
import java.util.Scanner;

public class Main {
    public static String findSum(String str1, String str2){
        if (str1.length() > str2.length()) {
            String temp = str1;
            str1 = str2;
            str2 = temp;
        }
        String str = "";
        int n1 = str1.length(), n2 = str2.length();
        StringBuilder sb1 = new StringBuilder(str1);
        StringBuilder sb2 = new StringBuilder(str2);
        sb1.reverse();
        sb2.reverse();

        int carry = 0;
        for (int i=0; i<n1; i++){
            int sum = ((sb1.charAt(i)-'0')+(sb2.charAt(i)-'0')+carry);
            str += (char)(sum%10 + '0');
            carry = sum/10;
        }
        for (int i=n1; i<n2; i++){
            int sum = ((sb2.charAt(i)-'0')+carry);
            str += (char)(sum%10 + '0');
            carry = sum/10;
        }
        if (carry > 0)
            str += (char)(carry+'0');
        StringBuilder result = new StringBuilder(str);
        result.reverse();
        return result.toString();
    }

    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        String A = input.next();
        String B = input.next();
        System.out.println(findSum(A, B));
    }
}

File: C/0003.c
#include <stdio.h>
#include <string.h>

char* findSum(char* str1, char* str2) {
    if (strlen(str1) > strlen(str2)) {
        char* temp = str1;
        str1 = str2;
        str2 = temp;
    }
    
    int n1 = strlen(str1);
    int n2 = strlen(str2);
    int maxLength = n2;
    
    int carry = 0;
    char* result = malloc((maxLength + 2) * sizeof(char));
    int index = maxLength;
    result[index] = '\0';
    
    for (int i = n1 - 1; i >= 0; i--) {
        int sum = (str1[i] - '0') + (str2[i + n2 - n1] - '0') + carry;
        result[--index] = sum % 10 + '0';
        carry = sum / 10;
    }
    
    for (int i = n2 - n1 - 1; i >= 0; i--) {
        int sum = (str2[i] - '0') + carry;
        result[--index] = sum % 10 + '0';
        carry = sum / 10;
    }
    
    if (carry > 0) {
        result[--index] = carry + '0';
    }
    
    return result + index;
}

int main() {
    char A[100], B[100];
    scanf("%s%s", A, B);
    printf("%s\n", findSum(A, B));
    return 0;
}